desc/geometry/curve.py

```diff
@@ -868,6 +891,7 @@ def Z(self, new):
                 + f"got {len(new)} Z values for {len(self.knots)} knots"
             )
 
+    @optimizable_parameter
```

Comment:
i don't think we want the knots themselves to be optimizable?

Reply:
Probably not, I have always been fixing them during optimization anyway. But not completely sure, @dpanici what do you think?

And if I do delete this line, then I should probably move `"knots"` from `params` to `transforms` right?

Reply:
Agreed, don't make this an optimizable parameter

Reply:
I deleted that line and put it in `transforms` instead